libnd4j/include/ops/declarable/platform/vednn/conv2d.cpp

```diff
@@ -221,46 +214,52 @@ PLATFORM_IMPL(conv2d, ENGINE_CPU) {
   auto status = res == VEDNN_SUCCESS ? sd::Status::OK : sd::Status::BAD_ARGUMENTS;
 #else
 
-  VEDA_HANDLE &handle = VEDA_HANDLE::getInstance();
+  VEDA_HANDLE &handle = VEDA::getInstance().getVEDA_HANDLE(0);
+  SCOPED_VEDA_CONTEXT scopedContext(handle.getDevice());
 
   auto func = handle.getFunctionByConstPtrName("vedaVednnConvolutionForwardAddBias");
 
   VEDAdeviceptr vIn, vW, vO;
-  VEDAdeviceptr vB;
+  VEDAdeviceptr vB = nullptr;
   size_t sizeIn = in->lengthOf() * in->sizeOfT();
   size_t sizeW = w->lengthOf() * w->sizeOfT();
   size_t sizeB = bias ? bias->lengthOf() * bias->sizeOfT() : 0;
   size_t sizeO = out->lengthOf() * out->sizeOfT();
 
-  VEDA(vedaMemAllocAsync(&vIn, sizeIn, 0));
-  VEDA(vedaMemAllocAsync(&vW, sizeW, 0));
-  if (bias) VEDA(vedaMemAllocAsync(&vB, sizeB, 0));
-  VEDA(vedaMemAllocAsync(&vO, sizeO, 0));
+  VEDA_CALL_THROW(vedaMemAllocAsync(&vIn, sizeIn, 0));
+  VEDA_CALL_THROW(vedaMemAllocAsync(&vW, sizeW, 0));
+  if (bias) VEDA_CALL_THROW(vedaMemAllocAsync(&vB, sizeB, 0));
+  VEDA_CALL_THROW(vedaMemAllocAsync(&vO, sizeO, 0));
 
-  VEDA(vedaMemcpyHtoDAsync(vIn, in->buffer(), sizeIn, 0));
-  VEDA(vedaMemcpyHtoDAsync(vW, w->buffer(), sizeW, 0));
-  if (bias) VEDA(vedaMemcpyHtoDAsync(vB, bias->buffer(), sizeB, 0));
+  VEDA_CALL_THROW(vedaMemcpyHtoDAsync(vIn, in->buffer(), sizeIn, 0));
+  VEDA_CALL_THROW(vedaMemcpyHtoDAsync(vW, w->buffer(), sizeW, 0));
+  if (bias) VEDA_CALL_THROW(vedaMemcpyHtoDAsync(vB, bias->buffer(), sizeB, 0));
 
-  // uint64_t res;
-  VEDA(vedaLaunchKernel(func, 0, VEDAstack(&paramIn, VEDA_ARGS_INTENT_IN, sizeof(paramIn)), vIn,
-                        VEDAstack(&paramFilter, VEDA_ARGS_INTENT_IN, sizeof(paramFilter)), vW,
-                        VEDAstack(&paramBias, VEDA_ARGS_INTENT_IN, sizeof(paramBias)), vB,
-                        VEDAstack(&paramOut, VEDA_ARGS_INTENT_IN, sizeof(paramOut)), vO,
-                        VEDAstack(&paramConv, VEDA_ARGS_INTENT_IN, sizeof(paramConv)), VEDNN_CONV_ALGORITHM_DIRECT));
+  // if(bias) sd_printf("%s\n","--------bias case--------");
```

Comment:
Mind getting rid of extra commented code down here or turn t his in to an sd_debug?